nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1373 @@
+       "      <th>4</th>\n",
+       "      <td>improve methods</td>\n",
+       "      <td></td>\n",
+       "      <td>keyword</td>\n",
+       "      <td>removed keyword extraction</td>\n",
+       "      <td>quick</td>\n",
+       "      <td>statistical</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5</th>\n",
+       "      <td>replacement gensim</td>\n",
+       "      <td></td>\n",
+       "      <td>pervious</td>\n",
+       "      <td>pervious quick</td>\n",
+       "      <td>-</td>\n",
+       "      <td>replacement</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>6</th>\n",
+       "      <td>methods generate</td>\n",
+       "      <td></td>\n",
+       "      <td>pervious quick</td>\n",
+       "      <td>quick tip</td>\n",
+       "      <td>methods</td>\n",
+       "      <td>gensim</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>7</th>\n",
+       "      <td>extraction module</td>\n",
+       "      <td></td>\n",
+       "      <td>pervious quick tip</td>\n",
+       "      <td>keyword extraction</td>\n",
+       "      <td>great</td>\n",
+       "      <td>graph</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>8</th>\n",
+       "      <td>pke comes</td>\n",
+       "      <td></td>\n",
+       "      <td>quick tip</td>\n",
+       "      <td>extraction module</td>\n",
+       "      <td></td>\n",
+       "      <td>great choice</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9</th>\n",
+       "      <td>pke replacement</td>\n",
+       "      <td></td>\n",
+       "      <td>tip</td>\n",
+       "      <td>gensim</td>\n",
+       "      <td></td>\n",
+       "      <td>wide range</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "              KeyBERT  ...                  TopicRank\n",
+       "0      methods unable  ...  keyword extraction module\n",
+       "1  keyword extraction  ...                        pke\n",
+       "2   languages provdes  ...          different methods\n",
+       "3      pervious quick  ...                 experiment\n",
+       "4     improve methods  ...                statistical\n",
+       "5  replacement gensim  ...                replacement\n",
+       "6    methods generate  ...                     gensim\n",
+       "7   extraction module  ...                      graph\n",
+       "8           pke comes  ...               great choice\n",
+       "9     pke replacement  ...                 wide range\n",
+       "\n",
+       "[10 rows x 6 columns]"
+      ]
+     },
+     "metadata": {
+      "tags": []
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Compare the keywords extracted by the given algorithms\n",
+    "selected_algorithms = ['KeyBERT', 'BART-based', 'TFIDF','YAKE', 'TextRank', 'TopicRank']\n",
+    "\n",
+    "for text in texts:\n",
```

Comment:
First thing that occurred to me is why BART gets only a couple of keywords. I saw Thomas D remark about that but couldn't see the explanation in the markdowns (maybe I just missed it though 🙈). Maybe put a line/comment about that?

Reply:
Do you mean for both of the BART-based extraction methods? Above I mention the that for one of the two models the maximum number of keywords is 3 because the dataset contains a maximum of 3 keywords per training document.

But since you both were confused I'll mention it here once more.